__tests__/index.test.js

```diff
@@ -191,4 +191,77 @@ describe('GET /api/articles/:article_id/comments', () => {
       expect(datesInMs).toBeSorted({descending: true})
     })
   })
+  it('should return an empty array if article exists, but no comments under it', () => {
+    return request(app)
+    .get('/api/articles/2/comments')
+    .expect(200)
+    .then((res) => {
+      const comments = res.body.comments
+      expect(comments).toEqual([])
+    })
+  })
+})
+
+describe('POST /api/articles/:article_id/comments', () => {
+  it('should add a comment onto an article and return the posted comment, given a valid user', () => {
+    const newComment = {
+      username: 'butter_bridge',
+      body: 'This is my opinion.'
+    }
+    return request(app)
+    .post('/api/articles/3/comments')
+    .send({newComment})
+    .expect(201)
+    .then((res) => {
+      const comment = res.body.comment
+      expect(comment).toMatchObject({
+        comment_id: expect.any(Number),
+        body: newComment.body,
+        article_id: 3,
+        author: 'butter_bridge',
+        votes: expect.any(Number),
+        created_at: expect.any(String)
+      })
+      expect(moment(comment.created_at, moment.ISO_8601).isValid()).toBe(true)
+    })
+  })
+  it('should return an error if the author of the comment is not listed in users', () => {
+    const newComment = {
+      username: 'Chris',
+      body: 'This is my opinion.'
+    }
+    return request(app)
+    .post('/api/articles/3/comments')
+    .send({newComment})
+    .expect(400)
```

Comment:
If it's the user not exist error, it should be 404 instead of 400. May need to check user exists or not